src/addons/retry.rs

```diff
@@ -0,0 +1,68 @@
+use bevy::prelude::*;
+
+use crate::{LocalCommand, Process, ProcessCompleted};
+
+#[derive(Debug, Component)]
+pub enum Retry {
+    Attempts(usize),
+}
+
+/// Retry failed processes according to the Retry component.
+///
+/// Processes without the Retry component are ignored.
+/// The Retry component is removed from the entity when retries are done.
+pub(crate) fn retry_failed_process(
+    mut commands: Commands,
+    query: Query<(&LocalCommand, &Retry)>,
+    mut process_completed_events: EventReader<ProcessCompleted>,
+) {
+    for process_completed_event in process_completed_events.read() {
+        if process_completed_event.exit_status.success() {
+            continue;
+        }
+        if let Ok((current_local_command, retry)) = query.get(process_completed_event.entity) {
+            match retry {
+                Retry::Attempts(retries) => {
+                    if let Some(mut entity_commands) =
+                        commands.get_entity(process_completed_event.entity)
+                    {
+                        // Remove Retry component and skip if no more retries left
+                        if *retries < 1 {
+                            entity_commands.remove::<Retry>();
+                            continue;
+                        }
+
+                        // Remove both the Process and LocalCommand
+                        entity_commands.remove::<(Process, LocalCommand)>();
+
+                        // Create new LocalCommand
+                        let mut new_local_command =
```

Comment:
If we don't despawn I think we can remove all this command replication code.
We should be able to just spawn a new process again, using the same function we use when adding a new command